File: packages/typescript/tests/system/frames.test.ts
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Frames", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { isMobile } = result;

      // Frames support is not implemented in Appium yet
      if (isMobile) skip("Frames support is not implemented in Appium yet");

      return result;
    };
  });

  it("nested frames", async ({ expect, setup }) => {
    const { al, $ } = await setup();

    await $.navigate("https://the-internet.herokuapp.com/nested_frames");

    await al.do("click MIDDLE text");

    const texts = await al.get("text from all frames");
    expect(texts).toEqual(["LEFT", "MIDDLE", "RIGHT", "BOTTOM"]);
  });

  it("cross origin iframe", async ({ expect, setup, skip }) => {
    const { al, $, driverId } = await setup();
    const assert = expect.assert;

    if (driverId !== "playwright")
      skip("Frames support is only implemented for Playwright currently");

    await $.navigate("cross_origin_iframe.html");

    await al.check("button 'Main Page Button' is present", { assert });
    await al.check("'Password' field is present", { assert });

    await al.do("type 'testuser' in the text input field");
    await al.check("Text input contains 'testuser'", { assert });

    await al.do("click Submit button");
    await al.check("'Form submitted' message is present", { assert });
  });
});

File: packages/python/examples/pytest/frames_test.py
from os import getenv

from pytest import mark


@mark.xfail(
    "appium" in getenv("ALUMNIUM_DRIVER", "selenium"),
    reason="Frames support is only implemented for Playwright and Selenium currently",
)
def test_nested_frames(al, navigate):
    navigate("https://the-internet.herokuapp.com/nested_frames")

    al.do("click MIDDLE text")
    assert al.get("text from all frames") == ["LEFT", "MIDDLE", "RIGHT", "BOTTOM"]


@mark.xfail(
    "playwright" != getenv("ALUMNIUM_DRIVER", "selenium"),
    reason="Frames support is only implemented for Playwright currently",
)
def test_cross_origin_iframe(al, navigate):
    navigate("cross_origin_iframe.html")

    al.check("button 'Main Page Button' is present")
    al.check("'Password' field is present")
    al.do("type 'testuser' in the text input field")
    al.check("Text input contains 'testuser'")
    al.do("click Submit button")
    al.check("'Form submitted' message is present")
